### PR TITLE
Adding the ability to make options requests

### PR DESCRIPTION
**Pull request by wmak**
- The assert method string is a bit long now, not sure if gapipy is
strict about line lengths so I've left it.

### CODE REVIEW

File: gapipy/request.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def __init__(self, client, resource, params=None, parent=None):
     def _request(self, uri, method, data=None, params=None, additional_headers=None):
         """Make an HTTP request to a target API method with proper headers."""
 
-        assert method in ['GET', 'POST', 'PUT', 'PATCH'], "Only 'GET', 'POST', 'PUT', and 'PATCH' are allowed."
+        assert method in ['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS'], "Only 'GET', 'POST', 'PUT', 'PATCH' and 'OPTIONS' are allowed."
```
Comment (bartek) [inline gapipy/request.py]: This is indeed a bit long. Really not an issue, but we could probably write it as such:

```
# place at top with other constants
ALLOWED_METHODS = ['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS']

assert method in ALLOWED_METHODS, "Only {} are allowed.".format(', '.join(ALLOWED_METHODS))
```

File: gapipy/resources/base.py [hunk 2]
```diff
@@ -24,6 +24,12 @@ def __init__(self, data, stub=False, client=None):
             self._uri = self._resource_name
         super(Resource, self).__init__(data, client)
 
+    @classmethod
+    def options(cls, client):
+        request = APIRequestor(client, cls)
+        response = request.options()
```
Comment (marz619) [inline gapipy/resources/base.py]: might as well do `return requests.options()` here

Comment (annakovale) [reply gapipy/resources/base.py]: or even shorter:
```
return APIRequestor(client, cls).options()
```

Comment (marz619) [reply gapipy/resources/base.py]: fair enough

Comment (wmak) [reply gapipy/resources/base.py]: @marz619 Also I'd say it's out of scope for this PR but `create` does this format too as its what I based this function on :wink:

**Comment by marz619** [comment]
@wmak can you update the README Queries section with the new `options` call info

File: gapipy/resources/base.py (hunk 2 above)
Comment (marz619) [reply gapipy/resources/base.py]: can leave it as is... `readability` is good too

**Pull request merged**